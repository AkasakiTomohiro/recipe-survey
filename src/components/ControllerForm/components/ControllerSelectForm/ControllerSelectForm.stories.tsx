/* eslint-disable react-hooks/rules-of-hooks */
/**
 * @file ControllerSelectForm.stories.tsx
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

import { FormProvider, useForm } from 'react-hook-form';

import { ControllerSelectForm } from './ControllerSelectForm';

import type { Meta, StoryObj } from '@storybook/react';

type Test = {
  name: 'GW' | 'MW'
}

const meta: Meta<typeof ControllerSelectForm> = {
  title : 'RecipeSurvey/components/ControllerForm/ControllerSelectForm',
  render: (args) => {
    const methods = useForm<Test>({
      defaultValues: {
        name: 'MW'
      }
    });

    return (
      <FormProvider {...methods} >
        <ControllerSelectForm {...args} />
      </FormProvider>
    );
  }
};

export default meta;

type Story = StoryObj<typeof ControllerSelectForm>;

export const Primary: Story = {
  args: {
    label: 'セレクト',
    name : 'name',
    list : ['MW', 'GW']
  }
};
