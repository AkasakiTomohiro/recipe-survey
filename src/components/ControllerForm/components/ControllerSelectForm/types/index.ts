/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

import { FieldPath } from 'react-hook-form';
import { IGridFormControlProps } from '~/components/ControllerForm';
import { RequiredFilterType } from '~/types';

import { SelectProps } from '@mui/material/Select';

export type IControllerSelectFormProps<T extends Record<string, any> = Record<string, any>> = SelectProps<any> & IGridFormControlProps & {

  /**
   * フォームの名前
   */
  label: string;

  /**
   * controlで管理しているオブジェクトのキーを指定
   */
  name: FieldPath<RequiredFilterType<T, string>>;

  /**
   * 選択肢
   */
  list: string[];
}