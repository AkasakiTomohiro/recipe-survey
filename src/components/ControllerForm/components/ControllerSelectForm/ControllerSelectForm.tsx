/**
 * @file ControllerSelectForm.tsx
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

import { useController, useFormContext } from 'react-hook-form';

import InputLabel from '@mui/material/InputLabel';
import MenuItem from '@mui/material/MenuItem';
import Select from '@mui/material/Select';

import { GridFormControl } from '../GridFormControl';
import { IControllerSelectFormProps } from './types';

/**
 * Primary UI component for user interaction
 *
 * @example
 *
 * 利用したいコンポーネントでジェネリックに型を定義する形でラップして利用すること
 *
 * const SelectForm = (props: IControllerSelectFormProps<XXXType>) => {
 *   return (<ControllerSelectForm {...props}/>);
 * };
 */
export const ControllerSelectForm = (props: IControllerSelectFormProps): JSX.Element => {
  const { control } = useFormContext();
  const { field } = useController({ name: props.name, control });
  return (
    <GridFormControl allGird={props.allGird}>
      <InputLabel id={props.name}>{props.label}</InputLabel>
      <Select
        {...field}
        labelId={props.name}
      >
        {
          props.list.map((m, i) => <MenuItem key={i} value={m}>{m}</MenuItem>)
        }
      </Select>
    </GridFormControl>
  );
};
