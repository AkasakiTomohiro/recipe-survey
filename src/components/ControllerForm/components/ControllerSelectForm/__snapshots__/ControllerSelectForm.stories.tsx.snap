// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ControllerForm/ControllerSelectForm Primary smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-13i4rnv-MuiGrid-root">
  <div class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root">
    <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-zx681a-MuiFormLabel-root-MuiInputLabel-root"
           data-shrink="true"
           id="name"
    >
      セレクト
    </label>
    <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-5a3kbg-MuiInputBase-root-MuiInput-root-MuiSelect-root">
      <div tabindex="0"
           role="combobox"
           aria-controls=":r1:"
           aria-expanded="false"
           aria-haspopup="listbox"
           aria-labelledby="name mui-component-select-name"
           id="mui-component-select-name"
           class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      >
        MW
      </div>
      <input aria-invalid="false"
             name="name"
             aria-hidden="true"
             tabindex="-1"
             class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
             value="MW"
      >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="ArrowDropDownIcon"
      >
        <path d="M7 10l5 5 5-5z">
        </path>
      </svg>
    </div>
  </div>
</div>
`;
